.gitlab/source_test_junit_upload/source_test_junit_upload.yml

```diff
@@ -1,8 +1,9 @@
 source_test_junit_upload:
+  rules:
   # Uncomment this to only run junit uploads on main/stable branches
-  # rules:
-  #   !reference [.on_main_or_release_branch]
-  when: always
+  # - !reference [.on_main_or_release_branch]
```

Comment:
Even if you uncomment this line, the job will always run (except on mergequeue branches). I think we should remove this comment altogether?